Support Terra Clasic

niodamian opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added support for the Terra Classic (columbus-5) network, including configuration, address prefix, and native token details.
  * Enabled MintScan API queries for the Terra network.
  * Included Terra Classic in preset and migration network lists for improved compatibility.
  * Expanded earn provider support to include Terra network and its native token symbol (LUNC).
<!-- end of auto-generated comment: release notes by coderabbit.ai -->
<!-- devin-review-badge-begin -->

---

<a href="https://app.devin.ai/review/onekeyhq/app-monorepo/pull/7983" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    <img src="https://static.devin.ai/assets/gh-open-in-devin-review-light.svg?v=1" alt="Open with Devin">
  </picture>
</a>
<!-- devin-review-badge-end -->

niodamian committed:
Update MintScanQuery.ts

niodamian committed:
Update settings.ts

niodamian committed:
Update IQuery.ts

niodamian committed:
Update presetNetworks.ts

niodamian committed:
Update networkIds.ts

revan-zhang commented:
![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/11cc2492-41e8-4138-939e-c9b40f669392?refs=2b743ed7ba9fb8c67ba54e1a1b57f3dcfa9208e2&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/11cc2492-41e8-4138-939e-c9b40f669392/license?refs=2b743ed7ba9fb8c67ba54e1a1b57f3dcfa9208e2&source=prChecksComment)

niodamian committed:
Update earnProvider.constants.ts

niodamian commented on packages/kit-bg/src/vaults/impls/cosmos/settings.ts:
```diff
@@ -94,6 +94,11 @@ const settings: IVaultSettings = {
+    'cosmos--columbus-5': {
+      addressPrefix: 'terra',
+      curve: 'secp256k1',
+      nativeTokenAddress: 'uluna',
```
https://github.com/cosmostation/chainlist/blob/main/chain/terra-classic/param_2.json

niodamian committed:
Update v4CoinTypeToNetworkId.ts

niodamian committed:
Update IQuery.ts

niodamian commented on packages/core/src/chains/cosmos/sdkCosmos/query/MintScanQuery.ts:
```diff
@@ -22,6 +22,7 @@ const NetworkIDMinScanMap: Record<string, string> = {
   [getNetworkIdsMap().cronosposchain]: 'cryptoorg',
   [getNetworkIdsMap().cosmoshub]: 'cosmos',
   [getNetworkIdsMap().akash]: 'akash',
+  [getNetworkIdsMap().terra]: 'terra',
```
i disabled mintScanQuery and enabled cosmwasmQuery

niodamian commented:
Also couldnt find where i could add CW20 support token for the network. 

Wanted to add Juris protocol logo and ca with gecko id for price injection * 


Terra Classic - https://github.com/cosmostation/chainlist/tree/main/chain/terra-classic 



Terra classsic main 2 native coins are LUNC and USTC 

[
    {
        "type": "native",
        "denom": "uluna",
        "name" : "Lunc",
        "symbol": "LUNC",
        "description": "The native staking coin of Terra Classic.",
        "decimals": 6,
        "image": "https://raw.githubusercontent.com/cosmostation/chainlist/master/chain/terra-classic/asset/lunc.png",
        "coinGeckoId": "terra-luna",
        "color": "#1E75EE"
    },
    {
        "type": "native",
        "denom": "uusd",
        "name" : "USTC",
        "symbol": "USTC",
        "description": "The USD stablecoin of Terra Classic.",
        "decimals": 6,
        "image": "https://raw.githubusercontent.com/cosmostation/chainlist/master/chain/terra-classic/asset/uusd.png",
        "coinGeckoId": "terrausd",
        "color": "#1E75EE"
    }, 
    
    And wanted to add as well the logos and infos about the CW20 tokens : https://github.com/cosmostation/chainlist/blob/main/chain/terra-classic/cw20_2.json

niodamian committed:
Update IQuery.ts

niodamian committed:
Update IQuery.ts

niodamian committed:
Update IQuery.ts

niodamian commented:
ok i did the changes bot said * how  can i import logo of juris ? cw20 token

niodamian committed:
Update IQuery.ts

niodamian commented on packages/core/src/chains/cosmos/sdkCosmos/query/IQuery.ts:
```diff
@@ -52,11 +52,12 @@ class QueryRegistry {
 const cosmwasmQuery = new CosmwasmQuery();
 queryRegistry.register(getNetworkIdsMap().juno, cosmwasmQuery);
-// queryRegistry.register(getNetworkIdsMap().terra, cosmwasmQuery); // terra2
+//queryRegistry.register(getNetworkIdsMap().terra, cosmwasmQuery);  // Terra Classic
```
done

originalix reviewed (changes requested):
Hi thanks for this pr. Supporting Terra Classic also requires server-side handling for token metadata, fee estimation, and other related features. Simply adding the network config on the client side won’t be sufficient for it to work properly.

So the review will be put on hold for now until server-side support is in place.

originalix commented on packages/kit-bg/src/migrations/v4ToV5Migration/v4CoinTypeToNetworkId.ts:
```diff
@@ -54,6 +55,7 @@ export const v4CoinTypeToNetworkId: Partial<
   [COINTYPE_BCH]: networkIdsMap.bch,
   [COINTYPE_XRP]: networkIdsMap.xrp,
   [COINTYPE_COSMOS]: networkIdsMap.cosmoshub,
+  [COINTYPE_TERRA]: networkIdsMap.terra,
```
Is `COINTYPE_TERRA` different from `COINTYPE_COSMOS` ?

niodamian replied:
removed COINTYPE_TERRA  since i added here : https://github.com/niodamian/app-monorepo/blob/x/packages/kit-bg/src/vaults/impls/cosmos/settings.ts#L97
<img width="459" height="285" alt="image" src="https://github.com/user-attachments/assets/bd41bc8e-48cb-4d42-b1dd-bf8acef60085" />

niodamian replied:
Terra uses more native coins LUNC / USTC / KRW should i add them or keep it only on uluna (LUNC )  ?

sidmorizon replied:
We currently only support one native token.

originalix commented on packages/shared/src/config/presetNetworks.ts:
```diff
@@ -2094,7 +2094,25 @@ const akash: IServerNetwork = {
+  'id': 'cosmos--columbus-5',
+  'impl': 'cosmos',
+  'isTestnet': false,
+  'logoURI': 'https://uni.onekey-asset.com/static/chain/terra.png',
```
Should upload terra logo.

niodamian replied:
added logo url from keplr wallet

niodamian commented:
> Hi thanks for this pr. Supporting Terra Classic also requires server-side handling for token metadata, fee estimation, and other related features. Simply adding the network config on the client side won’t be sufficient for it to work properly.
> 
> So the review will be put on hold for now until server-side support is in place.

I believe I added the fees of terra ...keplr wallet https://github.com/chainapsis/keplr-chain-registry/blob/main/cosmos/columbus.json

niodamian committed:
Update v4CoinTypeToNetworkId.ts

niodamian committed:
Update presetNetworks.ts

niodamian committed:
Update v4CoinTypeToNetworkId.ts

niodamian committed:
Update v4CoinTypeToNetworkId.ts

niodamian committed:
Update presetNetworks.ts

niodamian commented on packages/kit-bg/src/migrations/v4ToV5Migration/v4CoinTypeToNetworkId.ts:
```diff
@@ -8,7 +8,7 @@ import {
   COINTYPE_CFX,
   COINTYPE_CKB,
-  COINTYPE_COSMOS,
+  COINTYPE_COSMOS,  
   COINTYPE_DNX,
```
@originalix should i build the COINTYPE_TERRA ?

niodamian committed:
Update IQuery.ts

niodamian commented:
changed mintscan to cosmwasmQuery - currently mintscan doesnt support the chain or it supports the chain its not on the website cause its facing performance issues currently

niodamian committed:
Delete packages/components/svg/illus/terra.svg

niodamian committed:
Add files via upload

niodamian commented:
removed terra 2.0 logo and added terra 1.0 logo - 
On onekey site you guys had LUNC listed with Terra 2.0 logo on it - LUNC coin is on Terra classic aka Terra 1.0

niodamian commented:
> Hi thanks for this pr. Supporting Terra Classic also requires server-side handling for token metadata, fee estimation, and other related features. Simply adding the network config on the client side won’t be sufficient for it to work properly.
> 
> So the review will be put on hold for now until server-side support is in place.

https://github.com/chainapsis/keplr-wallet/blob/7aa7dbf06fa1f28ffce3a6b05ab0ce7d7141c3b3/apps/extension/src/config.ts#L1252 


Terra Classic uses all the natives coins but since  we closed all the stablecoins we currently using USTC / LUNC 


    feeCurrencies: [
      {
        coinDenom: "LUNC",
        coinMinimalDenom: "uluna",
        coinDecimals: 6,
        coinGeckoId: "terra-luna",
        gasPriceStep: {
          low: 28.325,
          average: 28.325,
          high: 28.325,
        },
      },
      {
        coinDenom: "USTC",
        coinMinimalDenom: "uusd",
        coinDecimals: 6,
        coinGeckoId: "terrausd",
        gasPriceStep: {
          low: 0.75,
          average: 0.75,
          high: 0.75,
        },
      },
      
 
 We got also a tax of 0.5% which applies on LUNC and USTC only CW20 tokens dont have a tax

niodamian commented:
originalix any update ? I already did some PRs on the other OneKey repos
